### PR TITLE
Decompile battle_3

### PR DESCRIPTION
**Pull request by DizzyEggg**
Credits to Cam for his port of battle3 which I ported. ABE (sadly) remains nonmatching and there's one ASM hax me and Cam had to use.

Oh, I also unintentionally added an empty hall of fame.c file, lol.

### CODE REVIEW

**Comment by yenatch** [comment]
well, why are hall_of_fame.c or hall_of_fame.h there? mistake? you can rebase it out.
```sh
git rebase -i a9a66b0^
```
the change `pick a9a66b0` to `edit a9a66b0`. then
```sh
git reset -p */hall_of_fame.*
# type y to unstage, n to leave it alone
git commit --amend
git rebase --continue
```

File: src/battle_3.c [hunk 1]
```diff
@@ -0,0 +1,6431 @@
...
+extern const u8 BattleScript_ColorChangeActivates[];
+extern const u8 BattleScript_RoughSkinActivates[];
+extern const u8 BattleScript_ApplySecondaryEffect[];
+extern const u8 BattleScript_CuteCharmActivates[];
+extern const u8 gUnknown_082DB68C[];
+extern const u8 BattleScript_SynchronizeActivates[];
+extern const u8 gUnknown_082DB4B8[];
+extern const u8 gUnknown_082DB4C1[];
+extern const u8 BattleScript_TraceActivates[];
+
```
Comment (ProjectRevoTPP) [inline src/battle_3.c]: why the extra line?

Comment (DizzyEggg) [reply src/battle_3.c]: Different set of battlescripts, called by the abilities/items. They should be externed in a sepereate battlescripts.h header anyway

Comment (ProjectRevoTPP) [reply src/battle_3.c]: put a comment elaborating this then.

**Pull request merged**